contracts/src/v0.7/dev/KeeperRegistryDev.sol

```diff
@@ -624,20 +661,149 @@ contract KeeperRegistryDev is
 
   /**
    * @notice calculates the minimum balance required for an upkeep to remain eligible
+   * @param id the upkeep id to calculate minimum balance for
    */
   function getMinBalanceForUpkeep(uint256 id) external view returns (uint96 minBalance) {
     return getMaxPaymentForGas(s_upkeep[id].executeGas);
   }
 
   /**
    * @notice calculates the maximum payment for a given gas limit
+   * @param gasLimit the gas to calculate payment for
    */
   function getMaxPaymentForGas(uint256 gasLimit) public view returns (uint96 maxPayment) {
     (uint256 gasWei, uint256 linkEth) = getFeedData();
     uint256 adjustedGasWei = adjustGasPrice(gasWei, false);
     return calculatePaymentAmount(gasLimit, adjustedGasWei, linkEth);
   }
 
+  /**
+   * @notice retrieves the permited incoming and outgoing registries from storage
+   */
+  function getPermittedRegistries() external view returns (address[] memory, address[] memory) {
+    uint256 idx;
+    address[] memory incoming = new address[](s_permittedIncomingRegistryAddresses.length());
+    address[] memory outgoing = new address[](s_permittedOutgoingRegistryAddresses.length());
+    for (; idx < incoming.length; idx++) {
+      incoming[idx] = s_permittedIncomingRegistryAddresses.at(idx);
+    }
+    for (idx = 0; idx < outgoing.length; idx++) {
+      outgoing[idx] = s_permittedOutgoingRegistryAddresses.at(idx);
+    }
+    return (incoming, outgoing);
+  }
+
+  /**
+   * @notice sets the approved registry addresses
+   * @dev this function wastes so much gas 🤦
+   */
+  function setPermittedRegistries(address[] calldata incoming, address[] calldata outgoing) external onlyOwner {
+    uint256 idx;
+    uint256 numIncoming = s_permittedIncomingRegistryAddresses.length();
+    uint256 numOutgoing = s_permittedOutgoingRegistryAddresses.length();
+    // delete old set members
+    for (idx = numIncoming; idx > 0; idx--) {
+      s_permittedIncomingRegistryAddresses.remove(s_permittedIncomingRegistryAddresses.at(idx - 1));
+    }
+    for (idx = numOutgoing; idx > 0; idx--) {
+      s_permittedOutgoingRegistryAddresses.remove(s_permittedOutgoingRegistryAddresses.at(idx - 1));
+    }
+    // add new ones
+    for (idx = 0; idx < incoming.length; idx++) {
+      s_permittedIncomingRegistryAddresses.add(incoming[idx]);
+    }
+    for (idx = 0; idx < outgoing.length; idx++) {
+      s_permittedOutgoingRegistryAddresses.add(outgoing[idx]);
+    }
+  }
+
+  function migrateUpkeeps(uint256[] calldata ids, address destination) external override {
```

Comment:
Adding this functionality for a single Id will reduce a lot of code. It will cause inconvenience during migration but not hard to overcome by writing scripts. 

Let's keep this in mind if we still run into contract size issues

Reply:
good point, definitely an opportunity to reduce size here

Reply:
another thing to consider here though... we want to be able to "auto migrate" users that opt in to that service. If we do this, it might be 1000s of upkeeps to migrate at a time. It might make our lives easier down the line to have a batch function.

Reply:
I'm for simplifying, especially with the current code size issues. I hope that we ship the next registry version before we get to thousands of upkeeps, and if we don't we will pay this cost once and then we can build a better migration then.

Reply:
I'm in support of simplifying. If it's okay with you guys, I propose we remove after we upgrade to v0.8 and start size-golfing?